npm/private/npm_translate_lock.bzl

```diff
@@ -277,36 +330,14 @@ def _gen_npm_imports(lockfile, root_package, attr):
         url = None
         if tarball:
             if _is_url(tarball):
-                if registry and tarball.startswith("https://registry.npmjs.org/"):
-                    url = registry + tarball[len("https://registry.npmjs.org/"):]
+                if registry and tarball.startswith(utils.npm_registry_url):
+                    url = registry + tarball[len(utils.npm_registry_url):]
                 else:
                     url = tarball
             else:
-                # pnpm 6.x may omit the registry component from the tarball value when it is configured
-                # via an .npmrc registry setting for the package. If there is a registry value, then use
-                # that as the prefix. If there isn't then prefix with the default npm registry value and
-                # suggest upgrading to a newer version pnpm.
                 if not registry:
-                    registry = "https://registry.npmjs.org/"
-
-                    # buildifier: disable=print
-                    if attr.warn_on_unqualified_tarball_url:
-                        print("""
-
-====================================================================================================
-WARNING: The pnpm lockfile package entry for {} ({})
-does not contain a fully qualified tarball URL or a registry setting to indicate which registry to
-use. Prefixing tarball url `{}`
-with the default npm registry url `https://registry.npmjs.org/`.
-
-If you are using an older version of pnpm such as 6.x, upgrading to 7.x or newer and
-re-generating the lockfile should generate a fully qualified tarball URL for this package.
-
-To disable this warning, set `warn_on_unqualified_tarball_url` to False in your
-`npm_translate_lock` repository rule.
-====================================================================================================
-
-""".format(name, version, tarball))
+                    (scope, _) = utils.parse_package_name(name)
+                    registry = "https://{}".format(registries[scope]) if scope in registries else utils.npm_registry_url
                 url = registry + tarball
```

Comment:
should make sure that `registry` ends with `/` here when concatenating the strings `url = registry + tarball` or else inject `/` separator if between them if it doesn't